extensions/INTEL/SPV_INTEL_masked_gather_scatter.asciidoc

```diff
@@ -0,0 +1,291 @@
+either 0 or a power of two. When the value is not 0, it is an assertion that
+each pointer value in 'PtrVector' has this alignment. The behavior is undefined if
+any pointer value in 'PtrVector' does not have this alignment. +
+ +
+'Mask' is a vector of boolean values with the same number of elements as the _InputVector_. It specifies which elements of
+'InputVector' should be scattered. +
+
+1+|Capability: +
+*{capability_name}*
+1+| 5 | {OpMaskedScatterINTEL_token}
+| '<id>' +
+'InputVector'
+| '<id>' +
+'PtrVector'
+| '<literal>' +
+'Alignment'
+| '<id>' +
+'Mask'
+|=====
+
+
+=== 3.42.11. Conversion Instructions
+
+Allow *OpTypeVector* to be _Result Type_ and type of an input for *OpConvertPtrToU*, *OpConvertUToPtr* instructions:
+Change the _Result Type_ of *OpConvertPtrToU* as follows:
+_Result Type_ must be a scalar or vector of _integer type_, whose Signedness operand is 0.
+
+Change the  _Pointer_ of *OpConvertPtrToU* as follows:
+_Pointer_ must be a _physical pointer type_ or a _vector_ with _physical pointer type_ _Component Type_.
+If the bit width of _Pointer_ is smaller than that of _Result Type_, the conversion zero-extends _Pointer_.
+If the bit width of _Pointer_ is larger than that of _Result Type_, the conversion truncates Pointer. For
+same bit width Pointer and _Result Type_, this is the same as *OpBitcast*.
+
+Change the _Result Type_ of *OpConvertUToPtr* as follows:
+_Result Type_ must be a _physical pointer type_ or a _vector_ with _physical pointer type_ _Component Type_.
+
+Change the _Integer Value_ of *OpConvertUToPtr* as follows:
+_Integer Value_ must be a scalar or vector of _integer type_, whose Signedness operand is 0.
+If the bit width of _Integer Value_ is smaller than that of _Result Type_, the
+conversion zero-extends _Integer Value_. If the bit width of _Integer Value_ is larger
+than that of _Result Type_, the conversion truncates Integer Value. For same width _Integer Value_ and _Result Type_,
+this is the same as *OpBitcast*.
+
+Allow _vector_ of _physical pointer type_ to be _Result Type_ and type of a _Pointer_ for
+*OpPtrCastToGeneric*, *OpGenericCastToPtr* and *OpGenericCastToPtrExplicit* instructions.
+
+Allow _vector_ of _physical pointer type_ to be _Result Type_ and type of an _Operand_ for *OpBitcast* instruction.
+
+
+=== 3.42.12. Composite Instructions
+
+Most of the Composite Instructions that are supposed to work with vector type do not have any restrictions about _Component Type_.
+This extension allows these instructions to operate on _vector_ of _physical pointer type_.
+
+Allow _physical pointer type_ to be a _Result Type_ of *OpVectorExtractDynamic*.
+
+
+=== Issues
+
+None
+
+Revision History
+----------------
+[cols="5,15,15,70"]
+[grid="rows"]
+[options="header"]
+|========================================
+|Rev|Date|Author|Changes
+|1|2022-08-19|Dmitry Sidorov|Initial revision
+|2|2023-09-05|Dmitry Sidorov|Prepare to ship
```

Comment:
The first published version is usually version 1, but I'm fine publishing version 2 if you are.